File: packages/core/expo-module/android/src/main/java/expo/modules/universaltextinput/UniversalTextInputModule.kt
package expo.modules.universaltextinput

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class UniversalTextInputModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("UniversalTextInput")

    View(UniversalTextInputView::class) {
      Prop("value") { view: UniversalTextInputView, value: String? ->
        view.setValue(value)
      }

      Prop("defaultValue") { view: UniversalTextInputView, value: String? ->
        view.setDefaultValue(value)
      }

      Prop("placeholder") { view: UniversalTextInputView, placeholder: String? ->
        view.setPlaceholder(placeholder)
      }

      Prop("editable") { view: UniversalTextInputView, editable: Boolean? ->
        view.setEditable(editable != false)
      }

      Prop("secureTextEntry") { view: UniversalTextInputView, secure: Boolean? ->
        view.setSecureTextEntry(secure == true)
      }

      Prop("multiline") { view: UniversalTextInputView, multiline: Boolean? ->
        view.setMultiline(multiline == true)
      }

      Prop("autoFocus") { view: UniversalTextInputView, autoFocus: Boolean? ->
        view.setAutoFocus(autoFocus == true)
      }

      Prop("dark") { view: UniversalTextInputView, dark: Boolean? ->
        view.setDarkMode(dark == true)
      }

      Prop("paddingHorizontal") { view: UniversalTextInputView, padding: Int? ->
        view.setPaddingHorizontal(padding ?: 0)
      }

      Prop("paddingVertical") { view: UniversalTextInputView, padding: Int? ->
        view.setPaddingVertical(padding ?: 0)
      }

      Events("onChangeText", "onInputFocus", "onInputBlur")
    }
  }
}

File: packages/core/expo-module/ios/UniversalTextInputModule.swift
import ExpoModulesCore

public class UniversalTextInputModule: Module {
  public func definition() -> ModuleDefinition {
    Name("UniversalTextInput")

    View(UniversalTextInputView.self) {
      Prop("value") { (view: UniversalTextInputView, value: String?) in
        view.setValue(value)
      }

      Prop("defaultValue") { (view: UniversalTextInputView, value: String?) in
        view.setDefaultValue(value)
      }

      Prop("placeholder") { (view: UniversalTextInputView, placeholder: String?) in
        view.setPlaceholder(placeholder)
      }

      Prop("editable") { (view: UniversalTextInputView, editable: Bool?) in
        view.setEditable(editable != false)
      }

      Prop("secureTextEntry") { (view: UniversalTextInputView, secure: Bool?) in
        view.setSecureTextEntry(secure == true)
      }

      Prop("multiline") { (view: UniversalTextInputView, multiline: Bool?) in
        view.setMultiline(multiline == true)
      }

      Prop("autoFocus") { (view: UniversalTextInputView, autoFocus: Bool?) in
        view.setAutoFocus(autoFocus == true)
      }

      Prop("dark") { (view: UniversalTextInputView, dark: Bool?) in
        view.setDarkMode(dark == true)
      }

      Prop("paddingHorizontal") { (view: UniversalTextInputView, padding: Int?) in
        view.setPaddingHorizontal(padding ?? 0)
      }

      Prop("paddingVertical") { (view: UniversalTextInputView, padding: Int?) in
        view.setPaddingVertical(padding ?? 0)
      }

      Events("onChangeText", "onInputFocus", "onInputBlur")
    }
  }
}
